libraries/core/src/main/java/com/paritytrading/philadelphia/FIXValue.java

```diff
@@ -91,6 +91,74 @@ public int length() {
         return length;
     }
 
+    /**
+     * Compare the string representation of this value to the specified
+     * character.
+     *
+     * @param x the character to compare the string representation of this
+     *   value against
+     * @return true if this value represents the same string as the specified
+     *   character, otherwise false
+     */
+    public boolean contentEquals(byte x) {
+        return length == 1 && bytes[offset] == x;
+    }
+
+    /**
+     * Compare the string representation of this value to the specified
+     * character sequence.
+     *
+     * @param x the character sequence to compare the string representation of
+     *   this value against
+     * @return true if this value represents the same string as the specified
+     *   character sequence, otherwise false
+     */
+    public boolean contentEquals(byte[] x) {
+        if (length != x.length)
+            return false;
+
+        for (int i = 0; i < length; i++) {
```

Comment:
What about `Arrays.equals`? Also possibly an intrinsic in JDK9.

Reply:
Unfortunately that cannot be used, because `bytes` is always going to be longer by at least one element (the SOH). Therefore the actual arrays are not equal.